test/e2e/config_reload_test.go

```diff
@@ -101,6 +102,68 @@ func TestConfigReloadOnClientAuthSecretChange(t *testing.T) {
 	// Skip the test if "ShareProcessNamespace" is not enabled.
 	f.Require(t, framework.ShareProcessNamespace)
 
+	// Create a NatsCluster resource with a single member, having configuration reloading enabled and using the secret above for client authentication.
+	ConfigReloadTestHelper(t, func(natsCluster *natsv1alpha2.NatsCluster, cas *v1.Secret) {
+		natsCluster.Spec.Auth = &natsv1alpha2.AuthConfig{
+			// Use the secret created above for client authentication.
+			ClientsAuthSecret: cas.Name,
+		}
+		natsCluster.Spec.Pod = &natsv1alpha2.PodPolicy{
+			// Enable configuration reloading.
+			EnableConfigReload: true,
+		}
+	})
+}
+
+// TestConfigReloadOnClientAuthSecretChange creates a secret containing authentication data for a NATS cluster.
+// This secret initially contains two users ("user-1" and "user-2") and the corresponding password.
+// Then, the test creates a NatsCluster resource that uses this secret for authentication, and makes sure that "user-1" can connect to the NATS cluster.
+// Finally, it removes the entry that corresponds to "user-1" from the authentication secret, and makes sure that "user-1" cannot connect to the NATS cluster anymore.
+func TestConfigReloadOnClientAuthFileChange(t *testing.T) {
+	// Skip the test if "ShareProcessNamespace" is not enabled.
+	f.Require(t, framework.ShareProcessNamespace)
+
+	ConfigReloadTestHelper(t, func(natsCluster *natsv1alpha2.NatsCluster, cas *v1.Secret) {
+		natsCluster.Spec.Auth = &natsv1alpha2.AuthConfig{
+			// Use the secret created above for client authentication.
+			ClientsAuthFile: "/authconfig/auth.json",
```

Comment:
For the `include` to work it has to be mounted at `/etc/nats-config/authconfig/` since the includes have to be relative to where the main config is located which is `/etc/nats-config/`.

Reply:
:scream: 

That's a huge caveat to add to the documentation.  Will definitely cause confusion.

Reply:
Agree... we'd have to cover with documentation for now.